hugr-py/src/hugr/_hugr.py

```diff
@@ -112,6 +115,7 @@ class NodeData:
     _num_inps: int = 0
     _num_outs: int = 0
     # TODO children field?
+    children: list[Node] = field(default_factory=list)
```

Comment:
Do we care about the order of children? Otherwise a set might be better for deleting nodes?

Reply:
yes, order of children is significant in HUGR in general. Deleting nodes does
not need to be efficient - HUGR rewriting efficiency is only a requirement of
the rust crate.